packages/bigquery/sodasql/dialects/bigquery_dialect.py

```diff
@@ -84,9 +84,12 @@ def sql_test_connection(self) -> bool:
         except Exception as e:
             raise Exception(f'Unable to list tables from: {dataset_id}. Exception: {e}')
 
-    def sql_tables_metadata_query(self, limit: str = 10, filter: str = None):
-        return (f"SELECT table_name \n"
-                f"FROM `{self.dataset_name}.INFORMATION_SCHEMA.TABLES`;")
+    def sql_tables_metadata_query(self, limit: int = None, filter: str = None):
+        sql = (f"SELECT table_name \n"
+               f"FROM `{self.dataset_name}.INFORMATION_SCHEMA.TABLES`")
+        if limit is not None:
+            sql += f"\n LIMIT {limit}"
```

Comment:
this limit part is repeating, perhaps it could be a Dialect method and we could do 
```
sql = ... " from ... {self.get_metadata_query_limit();"}
```
wdyt?
and the method could return None

Reply:
I think for the first version we probably don't need an extra method in the base class .. yet. My thoughts are we can also make the entire query configurable, so the API is not finalised yet - there is also 'unused' filter option.

Reply:
some context : https://github.com/sodadata/soda-sql/pull/242